splinter/include/splinter_ros/splinter1d.hpp

```diff
@@ -0,0 +1,28 @@
+#ifndef SPLINTER_ROS__SPLINTER1D_HPP_
+#define SPLINTER_ROS__SPLINTER1D_HPP_
+
+#include <memory>
+#include <vector>
+
+
+namespace splinter_ros
+{
+struct Splinter1dImpl;
+class Splinter1d
+{
+public:
+  explicit Splinter1d(const std::vector<double> & x,
+    const std::vector<double> & y);
+
+  ~Splinter1d();
```

Comment:
Drop destructor.

Reply:
I can't drop the destructor or I get a `sizeof` issue when compiling. I'm not sure I understand it, but the explanation is the compiler doesn't know how to allocate the forward declared `impl` pointer and needs an explicit destructor for `Splinter1d` defined in the .cpp in the same translational unit as the completed definition of `Splinter1dImpl`

see:
https://stackoverflow.com/questions/6012157/is-stdunique-ptrt-required-to-know-the-full-definition-of-t

Reply:
I could add a comment to that effect in the code

Reply:
Well that's unexpected :confused: